chalicelib/app_utils.py

```diff
@@ -771,6 +771,47 @@ def run_get_environment(environ):
     return process_response(response)
 
 
+def run_delete_environment(environ):
+    """
+    Removes the environ entry from the Foursight envs bucket. This effectively de-schedules all checks
+    but does not remove any data.
+    """
+    s3_connection = S3Connection('foursight-envs')
+    s3_resp = s3_connection.delete_keys([environ])
+    keys_deleted = s3_resp['Deleted']
+    if not keys_deleted:
+        response = Response(
+            body = {
+                'status': 'error',
+                'description': 'Invalid environment provided to delete',
```

Comment:
Is that really the only reason this could be? I'm not a big fan of descriptions that just make something up after-the-fact. I might say "Unable to comply with request."